packages/analytics-browser/src/config.ts

```diff
@@ -186,13 +186,13 @@ export const useBrowserConfig = async (
   // Step 1: Create identity storage instance
   const identityStorage = options.identityStorage || DEFAULT_IDENTITY_STORAGE;
   const cookieOptions = {
-    ...options.cookieOptions,
     domain:
       identityStorage !== DEFAULT_IDENTITY_STORAGE ? '' : options.cookieOptions?.domain ?? (await getTopLevelDomain()),
```

Comment:
```
      identityStorage !== DEFAULT_IDENTITY_STORAGE ? '' : options.cookieOptions?.domain ?? (await getTopLevelDomain()),
```

This change will also overwrite the domain.
I think we should move the domain after the  `...options.cookieOptions`, it will also capture the domain in the  options.cookieOptions?.domain at the end.

Reply:
i think we want to allow `domain` to be modified right?

Reply:
yes, 
but [here](https://github.com/amplitude/Amplitude-TypeScript/blob/main/packages/analytics-browser/src/config.ts#L44), we set the domain = '' by default.
The domain value can never be getTopLevelDomain().

Reply:
but that only applies when no `cookieOptions` object is passed to `BrowserConfig` which in our case we always pass an object right?

Reply:
i added tests to make sure that the values are set correctly: https://github.com/amplitude/Amplitude-TypeScript/pull/564

Reply:
I got it now after thinking more. 
if there is no cookiesOption.domain, the domain value will be set through     
``` identityStorage !== DEFAULT_IDENTITY_STORAGE ? '' : options.cookieOptions?.domain ?? (await getTopLevelDomain())```
I think it works fine.